lib/Form/ClientForm.php

```diff
@@ -25,6 +25,7 @@ class ClientForm extends Form
      * RFC3986. AppendixB. Parsing a URI Reference with a Regular Expression.
      */
     public const REGEX_URI = '/^[^:]+:\/\/?[^\s\/$.?#].[^\s]*$/';
+    public const REGEX_ALLOWED_ORIGIN_URL = '/http(s?)\:\/\/[^\s\/$.?#]+\.[^\s\/$.?#]+(\.[^\s\/$.?#]+)*$/i';
```

Comment:
URL regex are more readable if you use a delimiter other than '/' to indicate it is a regex. Then you need to do less escaping. e.g.  |https?\://|      maybe add a comment what this is checking. I get lost after the protocol part  :)

Reply:
Updated the regex a bit and added a short explanation.